src/app/campaign/flight/flight-form-control-container.component.ts

```diff
@@ -0,0 +1,168 @@
+import { Component, OnInit, OnDestroy, Input, Output, EventEmitter, ChangeDetectionStrategy } from '@angular/core';
```

Comment:
This ControlContainer component adds another layer to the flight form components. We have the flight.container, which subscribes to the ngrx selectors with async pipes so we don't have to track those subscriptions or store that state data anywhere extra. Following the flight.container is this ControlContainer component that has the parent form in its template and contains components that have form fields. The inner form field components inject ControlContainer so that their form events bubble up to the parent form.

Reply:
The validations all have to be set up with the parent form unless child form field components implement the ControlValueAccessor and Validator interfaces to encapsulate those validations. That step is something I would recommend for the more complex fields like zones so we don't have to stick validations up here sp far away from the fields they pertain to.